src/jobs/consumers/notifications/base-notification.consumer.ts

```diff
@@ -0,0 +1,39 @@
+import { Logger } from '@nestjs/common';
+import { Job } from 'bull';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Notification } from 'src/modules/notifications/entities/notification.entity';
+import { DeliveryStatus } from 'src/common/constants/notifications';
+import { NotificationsService } from 'src/modules/notifications/notifications.service';
+
+export abstract class BaseNotificationConsumer {
+  private readonly logger = new Logger(this.constructor.name);
+
+  constructor(
+    @InjectRepository(Notification)
+    protected readonly notificationRepository: Repository<Notification>,
+    protected readonly notificationsService: NotificationsService,
+  ) {}
+
+  async processNotificationQueue(
+    job: Job<number>,
+    sendNotificationFn: () => Promise<unknown>,
+  ): Promise<void> {
+    const id = job.data;
+    const notification = (await this.notificationsService.getNotificationById(id))[0];
+
+    try {
+      this.logger.log(`Sending notification with id: ${id}`);
+      const result = await sendNotificationFn();
```

Comment:
Why add Fn at the end of this function name?

Need to refactor and keep better names.

Check that youtube video and in future lets avoid such issues.

Reply:
This has been addressed in commit https://github.com/OsmosysSoftware/osmo-notify/pull/66/commits/847414985c1f399137354d4f56c214645c046230